vm/src/obj/objdict.rs

```diff
@@ -146,22 +147,32 @@ impl PyDict {
         !self.entries.is_empty()
     }
 
-    fn inner_eq(zelf: PyRef<Self>, other: &PyDict, vm: &VirtualMachine) -> PyResult<bool> {
-        if other.entries.len() != zelf.entries.len() {
+    fn inner_cmp(
+        zelf: PyRef<Self>,
+        other: PyDictRef,
+        size_func: fn(usize, usize) -> bool,
+        item: bool,
+        vm: &VirtualMachine,
+    ) -> PyResult<bool> {
+        if size_func(zelf.len(), other.len()) {
             return Ok(false);
         }
+        let other_cp = other.copy();
```

Comment:
I guess this is not nessessary if original code will also work for le or gt if we take smaller dict for loop instead of zelf. 
Then remained size of the other side of looped one will tell it is bigger than smaller one or not.

Reply:
Thank you for your advice. I revised it to iterate a smaller one.